src/v/cloud_storage/partition_manifest.cc

```diff
@@ -345,7 +372,11 @@ bool partition_manifest::advance_start_offset(model::offset new_start_offset) {
 }
 
 std::vector<segment_meta> partition_manifest::replaced_segments() const {
```

Comment:
I remember a discussion about including the the trailing segments (i.e. all segments below `_start_offset` in the return value of this function. Do we still want to do that (i'm +1 for it) ?